### PR TITLE
Fix: GriefDefender filter

### PR DESCRIPTION
**Pull request by ahdg6**
## Overview
Due to API changes, the old one can not work properly.
This PR fixes it and optimize related code.

## Submitter Checklist

- [x] Make sure you are opening from a topic branch (**/feature/fix/docs/ branch** (right side)) and not your main branch.
- [x] Ensure that the pull request title represents the desired changelog entry.
- [x] I tested my changes and approved their functionality.
- [x] I ensured my changes do not break other parts of the code
- [x] I read and followed the [contribution guidelines](https://github.com/IntellectualSites/.github/blob/main/CONTRIBUTING.md)

## Maintainer checklist

Before the changes are marked as `Ready for merge`: 

- [ ] There are at least 2 approvals from project developers or maintainers for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional modification steps from users, e.g. for permission changes, make sure the entry is added to the pull request description so it can be appended to the changelog.

### CODE REVIEW

File: worldedit-bukkit/build.gradle.kts
Comment (NotMyFault) [inline worldedit-bukkit/build.gradle.kts:154]: If GD provides flowpowered at runtime, there's no need for us to shade it in and relocate it under a package we should not touch. Nor should we shade fourth party dependencies.

File: worldedit-bukkit/src/main/java/com/fastasyncworldedit/bukkit/regions/GriefDefenderFeature.java
Comment (NotMyFault) [inline worldedit-bukkit/src/main/java/com/fastasyncworldedit/bukkit/regions/GriefDefenderFeature.java:42]: Which versions of GD did the old diff used to support and which versions does the new one support?

File: worldedit-bukkit/build.gradle.kts
Comment (dordsor21) [reply worldedit-bukkit/build.gradle.kts:154]: It's not being shaded? Presumably GD shades and relocates, this this is the same requirement as using the p2 bundled kyori stuff

Comment (ahdg6) [reply worldedit-bukkit/build.gradle.kts:154]: flow-math is included in GriefDefenderAPI, If we don't relocate it, the method `claim.getGreaterBoundaryCorner()` can not work correctly.

File: worldedit-bukkit/src/main/java/com/fastasyncworldedit/bukkit/regions/GriefDefenderFeature.java
Comment (ahdg6) [reply worldedit-bukkit/src/main/java/com/fastasyncworldedit/bukkit/regions/GriefDefenderFeature.java:42]: Now it support 2.0.0 and newer, I haven't tested it on 1.5.10 but may works.

File: worldedit-bukkit/build.gradle.kts
Comment (NotMyFault) [reply worldedit-bukkit/build.gradle.kts:154]: Sounds like something GD has to resolve rather than us working with hacky methods again.

> Presumably GD shades and relocates, this is the same requirement as using the p2 bundled kyori stuff

We do that there, because we actually work with adventure stuff and need to send messages and access them. The flowmath references have been removed.

Comment (bloodmc) [reply worldedit-bukkit/build.gradle.kts:154]: This is definitely an issue with GD's API that I'll have fixed in GD 2.1.0. I'll be including the relocated libraries in GD API 2.1.0 to match what is used at runtime so plugins no longer have to worry about relocation.
@ahdg6 Please update this PR after GD 2.1.0 is released.

File: worldedit-bukkit/src/main/java/com/fastasyncworldedit/bukkit/regions/GriefDefenderFeature.java
Comment (NotMyFault) [reply worldedit-bukkit/src/main/java/com/fastasyncworldedit/bukkit/regions/GriefDefenderFeature.java:42]: Good to know, I'll update the wiki instructions once merged.

File: worldedit-bukkit/build.gradle.kts
Comment (dordsor21) [reply worldedit-bukkit/build.gradle.kts:154]: @NotMyFault this can probably be merged with a note on the relocation? Rather than waiting for GD 2.1.0. Unless 2.1.0 is not long to be released @bloodmc ?

Comment (NotMyFault) [reply worldedit-bukkit/build.gradle.kts:154]:
```toml
griefdefender = "2.0.0-20211206.072436-12"
```
needs to be changed to a MAJOR.MINOR.PATCH version, otherwise renovate can't pick it up according to our configuration.

**Comment by ahdg6** [comment]
The latest commit work fine in my personal server.

**Review by NotMyFault — APPROVED** [review]
2.1.0-SNAPSHOT is good enough for our needs.
Thanks for your contribution.

**Review by dordsor21 — APPROVED** [review]

**Comment by NotMyFault** [comment]
This PR is ready to be merged, we will merge it in ~24 hours if there is no negative feedback, as documented by the [merge process](https://github.com/IntellectualSites/IntellectualSites/blob/main/docs/MAINTAINERS_CONTRIBUTORS.adoc#merge-process) documentation.

**Pull request merged**
